### PR TITLE
[test/integration] Add API logging

### PR DESCRIPTION
**Pull request by trasc**
#### What this PR does / why we need it:
Adds a way to record apiserver logs during integration tests.

The logs recording can be enabled by setting  `INTEGRATION_API_LOG_LEVEL` environment variable to a numerical valued grater then 0. 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: Makefile-test.mk [hunk 1]
```diff
@@ -68,7 +71,8 @@ test-integration: gomod-download envtest ginkgo mpi-operator-crd ray-operator-cr
 	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" \
 	KUEUE_BIN=$(PROJECT_DIR)/bin \
 	ENVTEST_K8S_VERSION=$(ENVTEST_K8S_VERSION) \
-	$(GINKGO) $(GINKGO_ARGS) -procs=$(INTEGRATION_NPROCS) --race --junit-report=junit.xml --output-dir=$(ARTIFACTS) -v $(INTEGRATION_TARGET)
+	API_LOG_LEVEL=$(INTEGRATION_API_LOG_LEVEL) \
+	$(GINKGO) $(GINKGO_ARGS) -procs=$(INTEGRATION_NPROCS) --junit-report=junit.xml --output-dir=$(ARTIFACTS) -v $(INTEGRATION_TARGET)
```
Comment (mimowo) [inline Makefile-test.mk:75]: I would suggest not to remove the `race` in this PR, it is not contributing to "Add API Logging".

Comment (trasc) [reply Makefile-test.mk]: sure, done

**Review by mimowo — COMMENTED** [review]
/lgtm

**Pull request merged**